build/lex.go

```diff
@@ -59,13 +56,23 @@ func ParseDefault(filename string, data []byte) (*File, error) {
 	return f, err
 }
 
+var buildRegexp = regexp.MustCompile(`(?:^|\.)(?:build|workspace)(?:$|\.)`)
+
+func isBuildFilename(basename string) bool {
+	basename = strings.ToLower(basename)
+	if isBuild, ok := BuildFilenames[basename]; ok {
+		return isBuild
+	}
+	return buildRegexp.MatchString(basename)
```

Comment:
This will return true for `build.bzl` and `workspace.bzl` files. For example:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/workspace.bzl
https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/docker/build.bzl

Reply:
That's a good point.

Is there a reason this matching is case-insensitive? I actually do have a bunch of BUILD-like files named `BUILD.bzl`, which I wanted to be picked up.

Reply:
It's case-insensitive because of case-insensitive file systems (e.g. on Windows or Mac).

Reply:
Those are case-insensitive but case-preserving, right? So, I would expect to be getting `BUILD.whatever` (in that casing) back from the filesystem.

Reply:
I've updated the PR to explicitly exclude `.bzl` and .sky` extensions from being treated as BUILD files.